src/contract/EvaluationOptionsEvaluator.ts

```diff
@@ -124,6 +124,12 @@ export class EvaluationOptionsEvaluator {
     if (manifestOptions) {
       const errors = [];
       for (const k in manifestOptions) {
+        // eslint-disable-next-line no-prototype-builtins
+        if (!userSetOptions.hasOwnProperty(k)) {
```

Comment:
this should prevent from sitations where someone is reading a contract with a manifest options that are not supported by the SDK version that is being used.